server/controllers/api/claim/publish/validateFileTypeAndSize.js

```diff
@@ -24,8 +25,24 @@ const validateFileTypeAndSize = (file) => {
       }
       break;
     default:
-      logger.debug('publish > file validation > unrecognized file type');
-      throw new Error('The ' + file.type + ' content type is not supported.  Only, image/jpg, image/png, image/gif, and video/mp4 content types are currently supported.');
+      // Handle STL check (file.type is empty for STLs and file.type cannot be reassigned)
+      file.isStl = false;
+      return new Promise((resolve, reject) => {
```

Comment:
This entangles synchronous validation with a threejs loader that uses async events.
Being a synchronous method, this shouldn't wrap with a promise either.